api/controllers/v2/github.ts

```diff
@@ -87,6 +141,138 @@ export const TriggerBuild = async (event: APIGatewayEvent): Promise<APIGatewayPr
       body: errMsg,
     };
   }
+
+  let body: WorkflowRunCompletedEvent;
+  try {
+    body = JSON.parse(event.body) as WorkflowRunCompletedEvent;
+  } catch (e) {
+    console.log('[TriggerBuild]: ERROR! Could not parse event.body', e);
+    return {
+      statusCode: 502,
+      headers: { 'Content-Type': 'text/plain' },
+      body: ' ERROR! Could not parse event.body',
+    };
+  }
+
+  if (body.workflow_run.conclusion != 'success')
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body:
+        'Build on branch ' +
+        body.workflow_run.head_branch +
+        ' is not complete and will not trigger smoke test site deployments ',
+    };
+
+  if (body.workflow_run.name != 'Deploy Staging ECS')
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body:
+        'Workflow ' +
+        body.workflow_run.name +
+        ' completed successfully but only Deploy Staging ECS workflow completion will trigger smoke test site deployments',
+    };
+
+  // if the build was not building master, no need for smoke test sites
+  if (body.workflow_run.head_branch != 'main' || body.repository.fork) {
+    console.log('Build was not on master branch in main repo, sites will not deploy as no smoke tests are needed');
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body:
+        'Build on branch ' +
+        body.workflow_run.head_branch +
+        ' will not trigger site deployments as it was not on master branch in upstream repo',
+    };
+  }
+
+  //automated test builds will always deploy in dotcomstg
+  const env = 'dotcomstg';
+
+  async function createAndInsertJob() {
+    for (const s in SMOKETEST_SITES) {
```

Comment:
You could refactor the `for` loop and use `SMOKTEST_SITES.map` to allow for concurrent requests. Right now, these will occur sequentially since the for loop will `await` at each iteration at various points. If we use the map method built into arrays, we can get back an array of `Promise<void>` which we then can call `await Promise.all()`. This will make things a bit quicker 👍

Reply:
Will do, thanks for the idea!

Reply:
I implemented this, please let me know if it's what you had in mind!